contentcuration/contentcuration/viewsets/feedback.py

```diff
@@ -48,6 +50,54 @@ class Meta:
         fields = BaseFeedbackSerializer.Meta.fields + BaseFeedbackEventSerializer.Meta.fields + BaseFeedbackInteractionEventSerializer.Meta.fields
 
 
+class RecommendationsInteractionEventSerializer(BaseFeedbackSerializer, BaseFeedbackInteractionEventSerializer):
+    recommendation_event_id = serializers.UUIDField()
+
+    class Meta:
+        model = RecommendationsInteractionEvent
+        fields = BaseFeedbackSerializer.Meta.fields + BaseFeedbackInteractionEventSerializer.Meta.fields + ['recommendation_event_id']
+
+    def create(self, validated_data):
+        return RecommendationsInteractionEvent.objects.create(**validated_data)
+
+    def update(self, instance, validated_data):
+        for attr, value in validated_data.items():
+            setattr(instance, attr, value)
+        instance.save()
+        return instance
+
+
+class RecommendationsEventSerializer(BaseFeedbackSerializer, BaseFeedbackEventSerializer):
+    content = serializers.JSONField(default=list)
```

Comment:
For the `time_hidden` field, we probably want to replace it with the server timestamp, if it's set in the request. For datetimes, we generally want to honor the server timestamp, because anything from the client side could be untrustworthy. For that same reason, this forces our hand in how we create these events. Since we have `created_at`, we must make a subsequent update call to set `time_hidden`.

Reply:
I see `time_hidden` is not nullable. We might want to change that. If for whatever reason, our tracking wasn't able to update `time_hidden` after the fact, then having it be null will allow us to exclude that data.